tests/functional/docs/test_generate.py

```diff
@@ -33,3 +66,23 @@ def test_select_limits_no_match(self, project):
         run_dbt(["run"])
         catalog = run_dbt(["docs", "generate", "--select", "my_missing_model"])
         assert len(catalog.nodes) == 0
+
+    def test_catalog_with_sources(self, project):
```

Comment:
Don't recall if you were there, but at the last brown bag we decided to do one test per class is almost all cases, in order to avoid all potential concurrency/ordering issues.